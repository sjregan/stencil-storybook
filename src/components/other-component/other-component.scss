@import '#$componentNameSelector';

// with shadow dom
:host {
  display: block;
  font-family: sans-serif;
}

#{$componentNameSelector} {
  font-size: ms(0);
  display: block;
  // block
  &__name {
    color: $color-green-400;

    #{$componentNameSelector}__example-for-inner-block {
      color: $color-red-400;
    }
  }
  // modifier
  &--modified {
    text-align: center;
  }
}
