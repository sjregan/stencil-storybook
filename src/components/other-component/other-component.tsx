import { Component, Prop, Event, EventEmitter, Host, h } from '@stencil/core';
import { format } from '../../utils/utils';
import { BEM, hostClasses, ComponentName } from '../../utils';

@Component({
  tag: 'other-component',
  styleUrl: 'other-component.scss',
  shadow: false,
})
export class OtherComponent {
  @ComponentName() componentName;

  /** First Prop */
  @Prop() first: string = 'Sner';

  /** Middle Prop */
  @Prop() middle: string;

  /** Last Prop */
  @Prop() last: string;

  /** Modifier Prop */
  @Prop() modifier: string = '';

  /**
   * Specify size limit in Kb
   * */
  @Prop() maxFileSize;

  /**
   * Yo im a multiline
   * can you see me?
   * */
  @Prop() multilineShizzle: boolean;

  /** Prevent */
  @Event() removeButtonResponse!: EventEmitter;

  private getText(): string {
    return format(this.first, this.middle, this.last);
  }

  render() {
    const cssClasses = hostClasses({
      componentName: this.componentName,
      modifier: this.modifier,
    });

    return (
      <Host
        class={`${cssClasses} dasdsa---dddsd container`}
        onClick={() => this.removeButtonResponse.emit({ test: 'test' })}
      >
        Other Component <span class={BEM(this, 'name')}> {this.getText()}</span>

        <my-component
            first="inner"
            middle="test"
            last="component"
        />

        <slot />
        <slot name="right" />
      </Host>
    );
  }
}
