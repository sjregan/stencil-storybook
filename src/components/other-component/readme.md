# other-component

<!-- Auto Generated Below -->
