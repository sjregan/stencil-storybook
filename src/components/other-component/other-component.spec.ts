import { newSpecPage } from '@stencil/core/testing';
import { setupSpec, getComponentNameFromFilename } from '../../utils/tests/setup';
import { OtherComponent } from './other-component';

const componentName = getComponentNameFromFilename(__filename);

describe(componentName, () => {
  it('default', async () => {
    const { component } = await setupSpec(OtherComponent);
    expect(component.outerHTML).toMatchSnapshot();
  });

  it('vs. default manually', async () => {
    const { root } = await newSpecPage({
      components: [OtherComponent],
      html: `<${componentName}></${componentName}>`,
    });
    expect(root).toMatchSnapshot();
  });
});
