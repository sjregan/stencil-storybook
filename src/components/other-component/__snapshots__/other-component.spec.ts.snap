// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`other-component default 1`] = `"<other-component class=\\"other-component dasdsa---dddsd container\\"><!----> Hello, World! I'm <span class=\\"other-component__name\\"></span>  </other-component>"`;

exports[`other-component vs. default manually 1`] = `
<other-component class="container dasdsa---dddsd other-component">
  <!---->
  Hello, World! I'm
  <span class="other-component__name"></span>
</other-component>
`;
